autoPyTorch/api/base_task.py

```diff
@@ -1333,6 +1340,215 @@ def fit_pipeline(self,
+                             "Please call the `search()` method of {} prior to "
+                             "fit_ensemble().".format(self.__class__.__name__))
+
+        if precision not in [16, 32, 64]:
+            raise ValueError("precision must be one of 16, 32, 64. Got {}".format(precision))
+
+        if self._logger is None:
+            self._logger = self._get_logger(self.dataset.dataset_name)
+
+        # Create a client if needed
+        if self._dask_client is None:
+            self._create_dask_client()
+        else:
+            self._is_dask_client_internally_created = False
+
+        ensemble_fit_task_name = 'EnsembleFit'
+        self._stopwatch.start_task(ensemble_fit_task_name)
+        if enable_traditional_pipeline:
+            if func_eval_time_limit_secs is None or func_eval_time_limit_secs > time_for_task:
+                self._logger.warning(
+                    'Time limit for a single run is higher than total time '
+                    'limit. Capping the limit for a single run to the total '
+                    'time given to Ensemble fit (%f)' % time_for_task
+                )
+                func_eval_time_limit_secs = time_for_task
+
+            # Make sure that at least 2 models are created for the ensemble process
+            num_models = time_for_task // func_eval_time_limit_secs
+            if num_models < 2:
+                func_eval_time_limit_secs = time_for_task // 2
+                self._logger.warning(
+                    "Capping the func_eval_time_limit_secs to {} to have "
+                    "time for a least 2 models to ensemble.".format(
+                        func_eval_time_limit_secs
+                    )
+                )
+        # We only want to run dummy predictions in case we want to build an ensemble
+        dummy_task_name = 'runDummy'
+        self._stopwatch.start_task(dummy_task_name)
+        self._do_dummy_prediction()
+        self._stopwatch.stop_task(dummy_task_name)
+
+        # ============> Run traditional ml
+        # We only want to run traditional predictions in case we want to build an ensemble
+        if enable_traditional_pipeline and self.ensemble_size > 0:
+            if STRING_TO_TASK_TYPES[self.task_type] in REGRESSION_TASKS:
+                self._logger.warning("Traditional Pipeline is not enabled for regression. Skipping...")
+            else:
+                traditional_task_name = 'runTraditional'
+                self._stopwatch.start_task(traditional_task_name)
+                elapsed_time = self._stopwatch.wall_elapsed(ensemble_fit_task_name)
+                time_for_traditional = int(
+                    time_for_task - elapsed_time
+                )
+                self._do_traditional_prediction(
+                    func_eval_time_limit_secs=func_eval_time_limit_secs,
+                    time_left=time_for_traditional,
+                )
+                self._stopwatch.stop_task(traditional_task_name)
```

Comment:
Could you separate this process as a function?

```suggestion
        def run_traditional_ml(self, task_name: str, runtime_limit: int) -> None:
            if STRING_TO_TASK_TYPES[self.task_type] in REGRESSION_TASKS:
                self._logger.warning("Traditional Pipeline is not enabled for regression. Skipping...")
            else:
                traditional_task_name = 'runTraditional'
                self._stopwatch.start_task(traditional_task_name)
                elapsed_time = self._stopwatch.wall_elapsed(task_name)
                time_for_traditional = int(runtime_limit)
                self._do_traditional_prediction(
                    func_eval_time_limit_secs=func_eval_time_limit_secs,
                    time_left=time_for_traditional,
                )
                self._stopwatch.stop_task(traditional_task_name)
```